### PR TITLE
implement and test task 3 GET /api/topics

### PR DESCRIPTION
**Pull request by Calor94**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const express = require("express");
+const { getTopics } = require("./controllers/get.controllers");
+const app = express();
+
+app.use(express.json());
```
Comment (mezzDavies) [inline app.js]: We don't need this just yet - it is good practice to only have code in place that is being used.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+const request = require("supertest");
+const app = require("../app");
+
+describe("/api/topics", () => {
+  test("GET /api/topics responds with status 200 and responds with an array of objects", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then((res) => {
+        res.body.topics.forEach((topic) => {
```
Comment (mezzDavies) [inline __tests__/app.test.js]: There should be an assertion here checking the length of the array is correct before the forEach. If we don't then we could get a false positive.

**Review by mezzDavies — CHANGES_REQUESTED** [review]
Looking good but we could consider a couple of errors here - a generic 404 that can be tested for and a 500 internal server error that cannot be tested for but should be in place.

Have a look [here](https://notes.northcoders.com/courses/js-back-end/error-handling) for examples of the way to handle errors 😄 

Please put these in place and then resubmit your PR. 👍

File: app.js [hunk 3]
```diff
@@ -2,8 +2,15 @@ const express = require("express");
...
-app.use(express.json());
-
 app.get("/api/topics", getTopics);
 
+app.all("*", (req, res) => {
+  res.status(404).send({ msg: "Page not found" });
+});
+
+app.use((err, req, res, next) => {
+  console.log(err);
```
Comment (HannahBurks) [inline app.js:12]: just a note to remove any console logs before pushing to github :)

**Review by HannahBurks — APPROVED** [review]
Nice work Callum 😄  Get it merged! 👏🏼

**Pull request merged**